### ISSUE TITLE
cSpell hangs and high cpu usage

### ISSUE THREAD

**Issue by CaselIT** [Jason-Rev/vscode-spell-checker#78]
Sometimes this extension hangs by showing a clock icon in the status bar instead of the usual information. The tooltip in the clock icon is `cSpell waiting...`. When it enters this state the extension stops working and a vscode process uses 100% of a cpu core. The only solution is to reload the vscode window.

Sorry but I do not have a way to reproduce it yet. I'll try to give you some more information. I'm using vscode with the python extension at the moment, but I do not know if it is limited to this case, since I've haven't found a way to reproduce it. Also this started to happen after the update to v1+ (I do not remember if the v1.0.0 had this problem)
It may be similar to #77 but in my case reloading the vscode window or closing it ends the cSpell process that uses the cpu.

**Comment by CaselIT** [comment]
Some other information.

Clicking on the clock icon on the status bar gives an error notification with `Unable to open 'Spell Checker Info': Cannot read property 'document' of undefined.`

**Comment by Jason3S** [comment]
Thank you.

This sounds like some sort of race condition.  I'll see if I can reproduce it.  Are you running on a Windows machine or something else?

Cheers,
Jason

**Comment by CaselIT** [comment]
Yes I'm using a windows 10 machine.
Thank you jason.

Let me know if I can be of any more help

**Comment by CaselIT** [comment]
If it can be useful, it seems to be connected with python, since in the last few days I've been using typescript and it the issue did not represent itself.

Is before I still have not found a way to reproduce it, so it may just be by chance.
Hope it helps

**Comment by Jason3S** [comment]
Thank you I had a similar issue when editing Python.  It went away when I disabled to extension.  I wonder if the extension is opening hidden documents (some extensions do, I had to add special code to prevent spell checking those documents).

**Comment by Jason3S** [comment]
I would like to fix this. But I need to consistent way to reproduce it.

I know this can be very frustrating for those of you that this is happening to. Please bear with me, I use VS Code and the spell checker everyday. I am not seeing this issue. I believe it is a real issue, because I had seen something like it a few months ago.

If you are seeing this issue consistently, please work with me to find a way to reproduce it. We will need to do a bit of sleuthing.

Questions:
* What extensions do you have installed? This is important to know, because this issue is most likely caused by the interaction between multiple extensions.
* What language are you using? Better yet, if you can point me to a github repo that I can clone, this would be ideal.
* When does it happen? On VS Code startup? Only after a while? etc.
* What OS and version are you using? If you are supper advanced, a vagrant box!

Please realize I don't have a lot of time to work on this.

**Comment by CaselIT** [comment]
Sadly (or luckily) it's been a while since I haven't had any other instance of this happening that I remember.
At the moment I'm mainly working with Typescript, while when I encountered this issue I was using python. So I think it's probable that the issue is linked to that.

Regarding your questions 
- When I was having this issue I had `streetsidesoftware.code-spell-checker`, `joelday.docthis`, `donjayamanne.python`, `eg2.tslint`. Only the python extension should be active on a python file though
- Python, but I could not find any way to reproduce reliably
- I've had this happen both shortly after vscode startup than after some time (hours). Could not find a reason why.
- Windows 10 preview. I do not remember which version.

Since at the moment neither me nor you can reproduce it, we could leave it on hold for the moment. Just a thought: I use two groups, one left and one right, so it may be linked somehow to this https://github.com/Jason-Rev/vscode-spell-checker/issues/88
Thanks for your time.

**Comment by CaselIT** [comment]
@Jason3S I haven't noticed this problem lately, so I'll close this issue. If the problem reappears I'll reopen it (or create a new issue).

Thanks for the effort you put into this extension.

**Comment by Jason3S** [comment]
Thank you.
